tests/all/utils/fixture.rs

```diff
@@ -17,66 +17,123 @@ pub struct Fixture {
     pub path: PathBuf,
 }
 
-/// Copy the given fixture into a unique temporary directory. This allows the
-/// test to mutate the copied fixture without messing up other tests that are
-/// also trying to read from or write to that fixture. The given path should be
-/// relative from the root of the repository, eg
-/// "tests/fixtures/im-from-brooklyn-the-place-where-stars-are-born".
-pub fn fixture<P>(fixture: P) -> Fixture
-where
-    P: AsRef<Path>,
-{
-    // Make sure that all fixtures end up sharing a target dir, and we don't
-    // recompile wasm-bindgen and friends many times over.
-    static SET_TARGET_DIR: Once = ONCE_INIT;
-    SET_TARGET_DIR.call_once(|| {
-        env::set_var(
-            "CARGO_TARGET_DIR",
-            Path::new(env!("CARGO_MANIFEST_DIR")).join("target"),
-        );
-    });
-
-    let fixture = fixture
-        .as_ref()
-        .canonicalize()
-        .expect("should canonicalize fixture path OK");
-    let dir = ManuallyDrop::new(tempfile::tempdir().expect("should create temporary directory OK"));
-    let path = dir.path().join("wasm-pack");
-    println!(
-        "wasm-pack: copying test fixture '{}' to temporary directory '{}'",
-        fixture.display(),
-        path.display()
-    );
+impl Fixture {
+    /// Create a new test fixture in a temporary directory.
+    pub fn new() -> Fixture {
+        // Make sure that all fixtures end up sharing a target dir, and we don't
+        // recompile wasm-bindgen and friends many times over.
+        static SET_TARGET_DIR: Once = ONCE_INIT;
+        SET_TARGET_DIR.call_once(|| {
+            env::set_var(
+                "CARGO_TARGET_DIR",
```

Comment:
I'd personally recommend just always setting this env var for subprocesses, but I think `Once` will probably do the trick as well. I'm always just sort of weirdly paranoid about changing env vars while running.

Reply:
FWIW, this part is just code motion and we were already doing this.

Reply:
👍 no worries, feel free to move on

Reply:
(Also, the actual subprocess spawning happens deep within wasm-pack, not in the tests :-/ I agree with the general principal, however!)